Fe-feat-atoms-jjy button 요소 개발

JunYoung7 opened the pull request:
### 📕 Issue Number

linked issue #16 
<br/>

### 📗 작업 내역

> 구현 내용 및 작업 했던 내역

- atomic/buttons 개발
   - NavButton
![image](https://user-images.githubusercontent.com/61405355/100297756-bf0efc00-2fd2-11eb-864d-92862d8b3a5f.png)

   - IconButton
![image](https://user-images.githubusercontent.com/61405355/100297776-cafabe00-2fd2-11eb-8cfd-cce4291d1388.png)


<br/>

### 📘 작업 유형

- 신규 기능 추가

<br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 아직 각 컴포넌트 요소들의 크기나 여백에 대해서는 정확히 정한 기준이 없습니다!
- 혹시 코드에 이상한 부분이나 불필요한 부분, 추가적으로 필요한 부분 있으면 의견 남겨주세요!!

<br/><br/>

JunYoung7 committed:
feat : NavButton 구현

# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지

- button에 선택적 props 적용을 위한 backgroundColorProps 분리
- 아이콘 클릭 시 페이지 이동을 위한 Router, Link 컴포넌트 추가
- 종 모양 아이콘 구현을 위한 svg 추가
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
feat : IconButton 구현

- '-' 버튼 모양을 위한 minus svg 파일 추가
- icon에 마우스를 올렸을 때 클릭을 유도하기 위한 icon에 cursor:pointer 추가
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
chore : IconButton logic 수정

- 재사용성 증가를 위해 SVG 파일을 children으로써 입력받을 수 있도록 변경
- type 호환을 위해 children.toString 사용
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

leejiwoo2002 reviewed (approved):
상단바 아이콘 잘 쓰겠습니다!

leejiwoo2002 commented on client/src/components/atoms/button/NavButton/NavButton.tsx:
```diff
@@ -0,0 +1,45 @@
+
+const Button = styled.button<backgroundColorProps>`
+  width: 50px;
+  height: 50px;
```
이 크기를 저랑 다르게 만들어서 UI가 이상하게 보일지도 모르겠네요
만들면서 의논해보겠습니다

JunYoung7 replied:
넵 지우님 만드실 때 같이 조정해나가면 될 거 같습니다

Zigje9 commented on client/src/components/atoms/button/NavButton/NavButton.tsx:
```diff
@@ -0,0 +1,45 @@
+const NavButton: React.FC<Props> = ({ backgroundColor, moveUrl }: Props) => {
+  return (
+    <Button backgroundColor={backgroundColor}>
+      <Router>
+        <Link to={moveUrl}>
+          <BellIcon src={Bell} alt="bell icon" />
+        </Link>
+      </Router>
```
react-router 를 이렇게 걸어 두는 군요! 적용못한게 있었는데 참고하겠습니다 :)

JunYoung7 replied:
넵 필요하신 부분에 활용해도 좋을 것 같습니다 :)

Zigje9 reviewed (approved):
고생하셨습니다~

do02reen24 commented on client/src/components/atoms/button/NavButton/NavButton.tsx:
```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components';
+import Bell from '@svg/bell.svg';
+import myColor from '@theme/color';
+import { BrowserRouter as Router, Link } from 'react-router-dom';
```
svg 아이콘을 호출하는 부분이 하단쪽에 있도록 호출 순서를 바꿔도 좋을 것 같아요.

JunYoung7 replied:
넵 반영하겠습니다!

do02reen24 commented on client/src/components/atoms/button/NavButton/NavButton.tsx:
```diff
@@ -0,0 +1,45 @@
+    <Button backgroundColor={backgroundColor}>
+      <Router>
+        <Link to={moveUrl}>
+          <BellIcon src={Bell} alt="bell icon" />
```
NavButton이 bell이랑 setting에서 재사용되는 요소기 때문에 Icon도 props 로 넘겨받고 스토리북에서 BellNavButton, SettingNavButton을 확인할 수 있도록 두 개 따로 생성해주시면 좋을 것 같아요.

do02reen24 replied:
svg 아이콘의 색깔을 변경할 수 있도록 svg를 `.tsx` 파일로 만들어주면 좋을 것 같아요. 나중에 상단바에 따라서 아이콘의 색상이 흰색이 될 수도 있기 때문에 변경가능하도록 해주시면 좋을 것 같습니다😀

do02reen24 commented on client/src/components/atoms/button/NavButton/NavButton.tsx:
```diff
@@ -0,0 +1,45 @@
+  width: 50px;
+  height: 50px;
+  border: 0;
+  background-color: ${(props) => props.backgroundColor};
```
Button의 background-color를 투명색으로 고정해줘도 될 것 같습니다. 상단바 색이 바뀌기 때문에 상위 div의 background-color를 보여주면 될 것 같아요

do02reen24 replied:
아이콘의 흰색 여부를 결정해야하기 때문에 backgroundColor를 넘겨주는 부분은 남겨도 될 것 같아요

do02reen24 reviewed (commented):
준영님 수고 많으셨습니다😀👍

JunYoung7 committed:
feat : Line(수평선) 구현

# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명
- 구간을 나누기 위한 수평선 component 구현
- 재사용성을 위한 linePercent, color props 화

JunYoung7 committed:
refactor : NavButton review 반영(수정)

# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

- navButton 재사용성 증가를 위한 return 형식 변경
- setting icon 사용을 위한 setting svg 추가
- svg 속성 값 제어를 위한 bell, setting svg파일 tsx화

JunYoung7 committed:
Merge branch 'develop' into FE-feat-atom-jjy
